client/src/redux/shopCart/cart-reducer.js

```diff
@@ -0,0 +1,44 @@
+import CartActionTypes from "./cart-types";
+import { addItemToCart, removeItemFromCart } from "./cart-utils";
+
+const INITIAL_STATE = {
+  hidden: true,
+  cartItems: []
+};
+
+const cartReducer = (state = INITIAL_STATE, action) => {
+  switch (action.type) {
+    case CartActionTypes.TOGGLE_CART_HIDDEN:
+      return {
+        ...state,
+        hidden: !state.hidden
+      };
+    case CartActionTypes.ADDED_ITEM:
```

Comment:
`ADDED_ITEM` is when user click the `Add item` button in product page. `INCREMENTED_ITEM_QUANTITY` is used in checkout page where the user can click to increase quantity of the items. Both uses the same logic